producer-kafka/src/main/java/io/stargate/producer/kafka/schema/SchemaRegistryProvider.java

```diff
@@ -78,21 +84,42 @@ public void createOrUpdateSchema(TableMetadata tableMetadata) {
     createOrUpdateValueSchema(tableMetadata);
   }
 
+  private Schema getSchemaBySubject(String subjectName) {
+    Integer schemaId = schemaIdPerSubject.get(subjectName);
+    if (schemaId == null) {
```

Comment:
I'm not sure I understand when this condition will hit.

Reply:
When there is a restart of the stargate node and the `schemaIdPerSubject` is empty.

Reply:
Ok, nice. In that case, this can be simplified. I will remove this code branch, and throw `IllegalStateException` if this invariant is not kept.